### PR TITLE
[stable/traefik] k8s ingressEndpoint default value support and modify liveness/readiness check

### PR DESCRIPTION
**Pull request by kozmagabor**
Could you please review it? @dtomcej 

Thanks

#### What this PR does / why we need it:
Modify liveness/readiness check to the specific endpoint
Add k8s ingressEndpoint default value support

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
    - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/templates/configmap.yaml [hunk 1]
```diff
@@ -100,18 +105,21 @@ data:
...
-        {{- if .Values.kubernetes.ingressEndpoint.hostname }}
-      hostname = {{ .Values.kubernetes.ingressEndpoint.hostname | quote }}
-        {{- end}}
-        {{- if .Values.kubernetes.ingressEndpoint.ip }}
-      ip = {{ .Values.kubernetes.ingressEndpoint.ip | quote }}
-        {{- end}}
-        {{- if .Values.kubernetes.ingressEndpoint.publishedService }}
-      publishedService = {{ .Values.kubernetes.ingressEndpoint.publishedService | quote }}
-        {{- end}}
+    {{- end}}
```
Comment (dtomcej) [inline stable/traefik/templates/configmap.yaml:115]: I agree with the logic change here.

File: stable/traefik/templates/configmap.yaml [hunk 2]
```diff
@@ -100,18 +105,21 @@ data:
...
+      {{- if .Values.kubernetes.ingressEndpoint.hostname }}
+    hostname = {{ .Values.kubernetes.ingressEndpoint.hostname | quote }}
       {{- end}}
+      {{- if .Values.kubernetes.ingressEndpoint.ip }}
+    ip = {{ .Values.kubernetes.ingressEndpoint.ip | quote }}
+      {{- end}}
+      {{- if .Values.kubernetes.ingressEndpoint.publishedService }}
+    publishedService = {{ .Values.kubernetes.ingressEndpoint.publishedService | quote }}
+      {{- end}}
+    {{- else }}
```
Comment (dtomcej) [inline stable/traefik/templates/configmap.yaml:127]: There should be no else though. If not defined, don't create an ingressEndpoint section.

File: stable/traefik/templates/configmap.yaml [hunk 3]
```diff
@@ -100,18 +105,21 @@ data:
...
+    [kubernetes.ingressEndpoint]
+      publishedService = "{{ .Release.Namespace }}/{{ template "traefik.fullname" . }}"
```
Comment (dtomcej) [inline stable/traefik/templates/configmap.yaml:129]: Not a fan of this personally. I think the note in the readme should be sufficient to prompt people what to use

File: stable/traefik/templates/deployment.yaml [hunk 4]
```diff
@@ -62,16 +62,18 @@ spec:
             cpu: "{{ .Values.cpuLimit }}"
             memory: "{{ .Values.memoryLimit }}"
         readinessProbe:
-          tcpSocket:
-            port: 80
+          httpGet:
+            path: /ping
```
Comment (dtomcej) [inline stable/traefik/templates/deployment.yaml:66]: `/ping` is a liveliness endpoint. Not sure it is appropriate for use in readiness in this case.

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/deployment.yaml (hunk 4 above)
Comment (kozmagabor) [reply stable/traefik/templates/deployment.yaml:66]: I think http check is better than tcp. The tcp check is shallow readiness check in this case. I found the similar way in nginx-ingress, gce-ingress charts.

File: stable/traefik/templates/configmap.yaml (hunk 3 above)
Comment (kozmagabor) [reply stable/traefik/templates/configmap.yaml:129]: Traefik cannot function if it's empty or not defined.

Comment (kozmagabor) [reply stable/traefik/templates/configmap.yaml:129]: @dtomcej What do you think? If it hasn't default value, this function cannot work properly.

**Comment by kozmagabor** [comment]
Could you please review it? @emilevauge @timoreimann

File: stable/traefik/Chart.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: traefik
-version: 1.46.0
+version: 1.47.0
 appVersion: 1.7.1
```
Comment (vdemeester) [inline stable/traefik/Chart.yaml]: 1.7.2 has been released :angel:

**Review by vdemeester — COMMENTED** [review]
SGTM

File: stable/traefik/Chart.yaml (hunk 5 above)
Comment (kozmagabor) [reply stable/traefik/Chart.yaml]: fixed :)

File: stable/traefik/values.yaml [hunk 6]
```diff
@@ -36,7 +36,7 @@ tolerations: []
 #   value: "value"
 #   effect: "NoSchedule|PreferNoSchedule|NoExecute(1.6 only)"
 ## Kubernetes ingress filters
-# kubernetes:
+kubernetes: {}
```
Comment (dtomcej) [inline stable/traefik/values.yaml:39]: This should be removed.

File: stable/traefik/templates/configmap.yaml (hunk 3 above)
Comment (dtomcej) [reply stable/traefik/templates/configmap.yaml:129]: I think that it should be configured, or disabled. No default provided.

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/configmap.yaml (hunk 3 above)
Comment (kozmagabor) [reply stable/traefik/templates/configmap.yaml:129]: Oh, I see. You want to disable this function.
Is it good for you if I create a flag like this?
```
    {{- else if .Values.kubernetes.ingressEndpoint.useDefaultServiceName }}
    [kubernetes.ingressEndpoint]
      publishedService = "{{ .Release.Namespace }}/{{ template "traefik.fullname" . }}"
    {{- end}}
```

**Pull request closed without merging**